connector/dialect.tdd

```diff
@@ -1,374 +1,133 @@
+      <formula>MAX(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='bool' />
     </function>
-    <function group='string' name='MID' return-type='str'>
-      <formula>(CASE WHEN %2 IS NULL THEN NULL ELSE SUBSTR(CAST(%1 AS TEXT), GREATEST(1,CAST(FLOOR(%2) AS INTEGER)), GREATEST(0,CAST(FLOOR(%3) AS INTEGER))) END)</formula>
-      <argument type='str' />
-      <argument type='real' />
+    <function group='aggregate' name='MAX' return-type='real'>
+      <formula>MAX(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='real' />
     </function>
-    <function group='string' name='MIN' return-type='str'>
-      <formula><![CDATA[(CASE
-	WHEN %1 IS NULL OR %2 IS NULL THEN NULL
-	WHEN %1 < %2 THEN %1
-	ELSE %2 END)]]></formula>
-      <argument type='str' />
-      <argument type='str' />
-    </function>
-    <function group='string' name='REGEXP_EXTRACT' return-type='str'>
-      <formula>(SELECT (REGEXP_MATCHES(%1, %2))[1])</formula>
-      <argument type='str' />
-      <argument type='str' />
-    </function>
-    <function group='string' name='REGEXP_EXTRACT_NTH' return-type='str'>
-      <formula>(SELECT (REGEXP_MATCHES(%1, %2))[%3])</formula>
-      <argument type='str' />
-      <argument type='str' />
-      <argument type='localint' />
-    </function>
-    <function group='string' name='REGEXP_MATCH' return-type='bool'>
-      <formula>(%1 ~ %2)</formula>
-      <argument type='str' />
-      <argument type='str' />
+    <function group='aggregate' name='MAX' return-type='int'>
+      <formula>MAX(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='int' />
     </function>
-    <function group='string' name='REGEXP_REPLACE' return-type='str'>
-      <formula>REGEXP_REPLACE(%1, %2, %3, &apos;g&apos;)</formula>
-      <argument type='str' />
-      <argument type='str' />
+    <function group='aggregate' name='MAX' return-type='str'>
+      <formula>MAX(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
       <argument type='str' />
     </function>
-    <function group='string' name='REPLACE' return-type='str'>
-      <formula>REPLACE(CAST(%1 AS TEXT), CAST(%2 AS TEXT), CAST(%3 AS TEXT))</formula>
-      <argument type='str' />
-      <argument type='str' />
-      <argument type='str' />
+    <function group='aggregate' name='MAX' return-type='datetime'>
+      <formula>MAX(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='datetime' />
     </function>
-    <function group='string' name='RIGHT' return-type='str'>
-      <formula>(CASE WHEN %2 &gt;= 0 THEN SUBSTR(CAST(%1 AS TEXT), (LENGTH(CAST(%1 AS TEXT)) - CAST(FLOOR(%2) AS INTEGER) + 1) ) ELSE NULL END)</formula>
-      <argument type='str' />
-      <argument type='real' />
+    <function group='aggregate' name='MAX' return-type='date'>
+      <formula>MAX(%1)</formula>
+      <unagg-formula>%1</unagg-formula>
+      <argument type='date' />
```

Comment:
what does it mean to have this given that MongoSQL doesn't actually support a date (as in date-without-time) type?

Reply:
Thanks for clarifying, I wasn't sure if the date was supported as an argument.  I have removed the functions with return type or argument type as `date`.